### PR TITLE
libobs: Reset swapchain NULL when display render end

### PR DESCRIPTION
**Pull request by wanhongqing123**
### Description

The initial state of the swap chain is empty, so after rendering a display, the state of the swap chain should be reset to the initial state.

### Motivation and Context

I want to make this code more logical.

### How Has This Been Tested?

The code has been tested on Windows and Mac.
### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: libobs/obs-display.c [hunk 1]
```diff
@@ -235,6 +235,7 @@ static inline bool render_display_begin(struct obs_display *display,
 static inline void render_display_end()
```
Comment (eric-bosch-appspace) [inline libobs/obs-display.c:235]: jghjghjg

**Review by eric-bosch-appspace — COMMENTED** [review]
test

**Comment by wanhongqing123** [comment]
你的邮件已收到，我会尽快的回复你

**Comment by RytoEX** [comment]
Please select a valid entry in the "Types of changes" section, and please fill out the Checklist.

**Comment by wanhongqing123** [comment]
> Please select a valid entry in the "Types of changes" section, and please fill out the Checklist.

done

File: libobs/obs-display.c [hunk 2]
```diff
@@ -273,11 +274,11 @@ void render_display(struct obs_display *display)
 
 		pthread_mutex_unlock(&display->draw_callbacks_mutex);
 
-		render_display_end();
-
 		GS_DEBUG_MARKER_END();
 
 		gs_present();
+
+		render_display_end();
```
Comment (jpark37) [inline libobs/obs-display.c:281]: It doesn't make sense to present a display before it's rendered.

File: libobs/obs-display.c [hunk 3]
```diff
@@ -235,6 +235,7 @@ static inline bool render_display_begin(struct obs_display *display,
 static inline void render_display_end()
 {
 	gs_end_scene();
+	gs_load_swapchain(NULL);
```
Comment (jpark37) [inline libobs/obs-display.c:238]: Philosophical debate. While it makes it less likely for the display to be rendered to by accident, the code also doesn't do anything for correct usage. I'd rather not, but it's up to Lain.

File: libobs/obs-display.c (hunk 2 above)
Comment (wanhongqing123) [reply libobs/obs-display.c:281]: render end This represents the original state

File: libobs/obs-display.c (hunk 3 above)
Comment (wanhongqing123) [reply libobs/obs-display.c:238]: original state of the swap is null， so render end should be null

**Comment by wanhongqing123** [comment]
@Lain-B

**Comment by Lain-B** [comment]
I feel like there's insufficient motivation and context behind this change. And I don't quite see a need for this change.

Additionally, in the future, please do not resolve comments left for you unless either the person leaving the comment agrees with your rational or explanation, or you've fixed whatever issue they may be commenting about.

**Pull request closed without merging**